src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -3692,7 +3692,13 @@ bool IdealLoopTree::do_remove_empty_loop(PhaseIdealLoop *phase) {
   // counted loop has limit check predicate.
   Node* phi = cl->phi();
   Node* exact_limit = phase->exact_limit(this);
-  Node* final_iv = new SubINode(exact_limit, cl->stride());
+
+  // There might have been a condition guarding the counted loop.
```

Comment:
There should always be a zero trip guard (ensured by code above). So, I would rephrase this to make it more clear that we always need to do the pin on this guard (or remove this phrase completely).

Reply:
Thx. I rephrased the comment